src/components/OpenSeadragonViewer.js

```diff
@@ -73,7 +72,6 @@ export class OpenSeadragonViewer extends Component {
 
     this.osdCanvasOverlay = new OpenSeadragonCanvasOverlay(this.viewer);
     this.viewer.addHandler('update-viewport', this.onUpdateViewport);
-    this.viewer.addHandler('viewport-change', this.onViewportChange);
```

Comment:
I don't think `update-viewport` covers all the cases where we need to collect data. At least, when I toggled back and forth between two different windows, I could pretty reliably get the viewer to jump around.

Reply:
I've pushed up an additional commit here that I think addresses the problem. Let me know what you think.